src/headless/tasks/test/test_celery_task.py

```diff
@@ -393,3 +402,139 @@ def test_clean_metadata(self):
         self.assertTrue(isinstance(metadata['url'], basestring))
         self.assertTrue(
             isinstance(metadata['extra_keywords']['url'], basestring))
+
+    @unittest.skipUnless(
+        ast.literal_eval(os.environ.get('TASK_ALWAYS_EAGER', 'True')),
+        'This task only makes sense when it was tested on the same thread.'
+        'It will be skipped on async Celery test, because the validity check'
+        ' was on different QGIS Instance, thus will always be success '
+        '(and useless) on that setup.'
+    )
+    def test_layer_registry_empty_on_subsequent_run(self):
+        """Test layer registry were properly cleaned up
+
+        More about this test.
+        We are testing that a subsequent run of analysis or multi exposure
+        analysis or report generation will properly cleanup QGIS Layer
+        registry.
+        This is because the same QGIS instance is used on the same thread,
+        but different if it is on a different thread. We have to make sure
+        each new analysis or report generation is a fresh one.
+
+        Assuming other unittests were checking on each task validity, then
+        this method will only test the state of layer registry between each
+        task to avoid test duplication.
+        """
```

Comment:
Long test of sequentially synchronouos multiple layer registry validity check.
Doesn't check the result of each task though, because it is checked on another method.